### PR TITLE
Android SDK basic sample project and symlinked SDK Mentat binaries

### LINKED ISSUE #721: Automate packaging of rust artifacts in the Android SDK
Follow-up to #718 and #614. Ideally, no manual actions should be required to keep `libmentat_ffi.so` up-to-date, both while working on Rust code (trivially achieved locally via a symlink to .../target/...?) and when publishing the SDK.

`build --release` version should be used for bundling within the SDK.

### PR DESCRIPTION
**Pull request by grigoryk**
Changes:
- Adds a sample android app which uses mentat SDK and opens a mentat db on launch
- Changes jniLibs/**/libmentat_ffi.so files within the SDK to be symlinks instead of binaries

This makes SDK development flow a bit nicer and provides a simple example project we can iterate on to test and showcase SDK features. Something like this:
- Compile `mentat_ffi` for all targets via `scripts/android_build.sh` (symlinks now resolve)
- Open the Mentat android project
- Make changes, try the sample project, make sure stuff works and APIs are all right, etc
- Make more changes to mentat-proper. `scripts/android_build.sh x86`, and try them out in an emulator. Simply relaunching the sample project should pick up the new version of the symlinked binary.
- Eventually, compile the final binaries `scripts/android_build.sh`, bump `ext.library.version` in `build.gradle` and upload to bintray: `gradle bintrayUpload`

This largely fixes #721 ("release" binaries are symlinked, so non-ideal for local dev).

### CODE REVIEW

File: sdks/android/Mentat/samples/basic/build.gradle [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation project(':library')
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation "com.android.support:appcompat-v7:27.1.1"
```
Comment (ncalexan) [inline sdks/android/Mentat/samples/basic/build.gradle]: Consider lifting this.

**Comment by ncalexan** [comment]
> This makes SDK development flow a bit nicer and provides a simple example project we can iterate on to test and showcase SDK features. Something like this:
> 
>     Compile mentat_ffi for all targets via scripts/android_build.sh (symlinks now resolve)
>     Open the Mentat android project
>     Make changes, try the sample project, make sure stuff works and APIs are all right, etc
>     Make more changes to mentat-proper. scripts/android_build.sh x86, and try them out in an emulator. Simply relaunching the sample project should pick up the new version of the symlinked binary.
>     Eventually, compile the final binaries scripts/android_build.sh, bump ext.library.version in build.gradle and upload to bintray: gradle bintrayUpload
> 
> This largely fixes #721 ("release" binaries are symlinked, so non-ideal for local dev).

Can we get this fleshed out a little bit into a README somewhere in the FFI directory?

**Comment by grigoryk** [comment]
I'll add something useful to README regarding the flow.

I'll try to make the `debug` and `release` split work.

I agree regarding symlinking... I think it's a reasonable approach, given our audience (ourselves). Adding a caveat to README might help a bit.

**Comment by grigoryk** [comment]
Added some basic docs.

I'm going to punt the debug/release split to https://github.com/mozilla/mentat/issues/733. It's all right for now.

**Pull request merged**